src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngine.java

```diff
@@ -334,6 +305,42 @@ private static List<Event> makeFinalListOfEventsInOrder(final AssemblyResultSet
         return eventsInOrder;
     }
 
+    /**
+     * Partition events into clusters that must be considered together, either because they overlap or because they belong to the
+     * same mutually exclusive pair or trio.  To find this clustering we calculate the connected components of an undirected graph
+     * with an edge connecting events that overlap or are mutually excluded.
+     */
+    private static List<EventGroup> getEventGroupClusters(List<Event> eventsInOrder, List<List<Event>> disallowedPairsAndTrios) {
```

Comment:
I like this... you should keep in mind that you will need to do a similar grouping operation based off of STR adjacency in the future as well... (looks easy enough to add here...)